gpflow/conditionals.py

```diff
@@ -122,6 +232,267 @@ def base_conditional(Kmn, Kmm, Knn, f, *, full_cov=False, q_sqrt=None, white=Fal
     return fmean, fvar
 
 
+def independent_latents_conditional(Kmn, Kmm, Knn, f, *, full_cov=False, full_cov_output=False, q_sqrt=None,
+                                    white=False):
+    """
+    :param Kmn: M x L x N x P
+    :param Kmm: L x M x M
+    :param Knn: N x P  or  N x N  or  P x N x N  or  N x P x N x P
+    :param f: data matrix, M x L x R
+    :param q_sqrt: R x L x M x M  or  M x L x R
+    :return: N x R x P  ,  N x R x P x P
+    """
+    # TODO: Allow broadcasting over L if priors are shared?
+    # TODO: Change Kmn to be L x M x N x P? Saves a transpose...
+    R = tf.shape(f)[2]
+    M, L, N, P = [tf.shape(Kmn)[i] for i in range(Kmn.shape.ndims)]
+
+    Lm = tf.cholesky(Kmm)  # L x M x M
+
+    # Compute the projection matrix A
+    Kmn = tf.reshape(tf.transpose(Kmn, (1, 0, 2, 3)), (L, M, N * P))
```

Comment:
just checking- it's not possible to avoid the transpose by having Kmn computed differently ?

Reply:
Yes, there's a TODO for that up top.